protocol/pubsub/v2/message.go

```diff
@@ -119,15 +120,13 @@ func (m *Message) GetExtension(name string) interface{} {
 	return m.internal.Attributes[prefix+name]
 }
 
-// Finish marks the message to be forgotten.
-// If err is nil, the underlying Pubsub message will be acked;
-// otherwise nacked and return the error.
+// Finish marks the message to be forgotten and returns the provided error without modification.
+// If err is nil or of type protocol.ResultACK the PubSub message will be acknowledged, otherwise nack-ed.
```

Comment:
I've updated the pubsub doc according to your comment.
About the error returned by Finish, I agree this is "weird" currently.
But note that the error returned is actually used to perform some logging...
I suggest we open another Issue to discuss these topics:
- do not forward the error in Finish
- use error only for real acknowledgement error
- add an eventual "status" result to finish result to share with "ceClient" (logs, observability, ...)